src/cells/model.ts

```diff
@@ -429,6 +438,15 @@ class CodeCellModel extends BaseCellModel implements ICodeCellModel {
   }
 
   /**
+   * Clear the cell state.
+   */
+  clear(): void {
+    this.output.clear(false);
+    this.executionCount = null;
+    this.input.prompt = 'In [ ]:';
```

Comment:
`this.input.clear`? Might as well make the input model also have a say in what `clear` means, just like the output model has.